index.bs

```diff
@@ -1877,6 +1877,37 @@ If the bytes in the buffer have a natural intepretation
 as one of the other TypedArray types, provide that instead.
 For example, if the bytes represent Float32 values, use a {{Float32Array}}.
 
+<h3 id="allow-microtask-switches">A Promise represents a value, not a callback</h3>
+
+Avoid requiring that something needs to happen synchronously within
+a promise resolution or rejection callback.
+
+Authors take for granted that an `async` function can be wrapped
+in another that appends synchronous code at the end e.g. to inspect state
+using try/catch. But the `await` queues a microtask:
+<div class="example">
+```js
+async function fooWrapper() {
+  const foo = await platform.foo();
+  console.log("success");
+  return foo;
+}
+
+(async () => {
+  const foo = await fooWrapper();
+  foo.bar(); // on the same task, but not the same microtask that logged "success"
+})();
+```
+</div>
+
+To not interfere with this pattern, the lowest recommended restriction is "same task".
```

Comment:
```suggestion
To allow this pattern, the specification for bar() can say that it must be called in the same task as the one where the promise is returned, but cannot say that it must be called in the same microtask.
```

Trying to make text that I will understand the next time I read it.

Reply:
I've incorporated this in a new version. PTAL